### DISCUSSION TITLE
Prometheus Config to scrape Multiple Kubernetes Cluster- Design question

### DISCUSSION THREAD

**Discussion by dinesh4747** [thanos-io/thanos#3547] [Questions & Answers]
Hi Guys,

We wanted to have prometheus which would scrape metrics across more than 1 cluster different kubernetes cluster

As far as I understand using kubernetes_sd_configs we just happened to configure single master-api service till date and But with Multi cluster approach how do we devise a mechanism to discover 50 clusters and in that case How do we pass the list of multiple api-server url on the same kubernetes_sd_configs that should help us scrape across N+1 number of clusters 

I do understand that we can have 50 kubernetes_sd_configs to achieve it - But doesn’t that look bit redundant

**Comment by GiedriusS** [comment]
This question seems related more to Prometheus itself. Indeed that's the approach. If you have so many k8s clusters then it's probably safe to say that you ought to invest some time into using some kind of templating engine for your Prometheus configuration (:
